core/src/consensus/fork_choice.rs

```diff
@@ -1,18 +1,36 @@
 use {
-    crate::{
-        consensus::{
-            latest_validator_votes_for_frozen_banks::LatestValidatorVotesForFrozenBanks,
-            progress_map::ProgressMap, SwitchForkDecision, Tower,
-        },
-        replay_stage::HeaviestForkFailures,
+    crate::consensus::{
+        latest_validator_votes_for_frozen_banks::LatestValidatorVotesForFrozenBanks,
+        progress_map::ProgressMap, SwitchForkDecision, Tower,
     },
     solana_runtime::{bank::Bank, bank_forks::BankForks},
+    solana_sdk::slot_history::Slot,
     std::{
         collections::{HashMap, HashSet},
         sync::{Arc, RwLock},
     },
 };
 
+#[derive(PartialEq, Eq, Debug)]
+pub enum HeaviestForkFailures {
```

Comment:
conceptually I think these should belong in replay, as they are not related to fork choice but failures from either the tower or replay checks.